tests/testthat/test-weighting.R

```diff
@@ -20,12 +20,17 @@ with_mock_HTTP({
             "/api/datasets/3/preferences/",
             '{"weight":null}')
     })
-
+    
     test_that("Errors are properly handled when setting weight", {
         expect_error(weight(newds) <- "a",
             "Weight must be a Variable or NULL")
         ## test error handling when trying to set non-numeric (need backend?)
     })
+    
+    test_that("weightVariables function for dataset with no weight", {
+        expect_is(weightVariables(oldds), "VariableCatalog")
+    })
```

Comment:
Good start, but we need to assert more about the weight variables catalog than just its existence. If there are no weights in the dataset, assert that it is length 0.